besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -2240,6 +2242,16 @@ private MiningParameters getMiningParameters() {
   private DataStorageConfiguration getDataStorageConfiguration() {
     if (dataStorageConfiguration == null) {
       dataStorageConfiguration = dataStorageOptions.toDomainObject();
+
+      final ImmutableDataStorageConfiguration.Builder theBuilder =
+          ImmutableDataStorageConfiguration.builder().from(dataStorageConfiguration);
+      if (syncMode == SyncMode.FULL
+          && dataStorageOptions.toDomainObject().getDataStorageFormat()
+              == DataStorageFormat.BONSAI) {
+        logger.warn("Forcing bonsai-limit-trie-logs-enabled to false, since sync mode is FULL");
+        theBuilder.bonsaiLimitTrieLogsEnabled(false);
+      }
+      dataStorageConfiguration = theBuilder.build();
```

Comment:
this looks like a potential merge/rebase artifact ?

Reply:
This change was intentional, although it is a change to another PR that recently merged (https://github.com/hyperledger/besu/pull/7193) which stopped Besu starting if `FULL` and `Bonsai` were enabled together. I've tidied up the var name I used and added a comment explaining that we're just disabling `bonsai-limit-trie-logs-enabled` if that combination of sync mode and DB is being used.

Reply:
I would prefer this to be in a different PR since it isn't related to BFT snapsync.  It isn't clear to me that we want to entirely prevent trie log pruning if the sync mode is full.  Linea in particular has some atypical configurations.  

Personally I think _ALL_ of these features should be wrapped in profiles rather than trying to have one-default-fits-all.  @matkt @non-fungible-nelson